packages/apollo-language-server/src/project/client.ts

```diff
@@ -385,12 +385,13 @@ export class GraphQLClientProject extends GraphQLProject {
       (!clientSchemaDirectives || !clientSchemaDirectives.length)
     )
       return current;
+    console.warn({ clientOnlyDirectives, clientSchemaDirectives });
 
     const filtered = Object.create(null);
     for (const operationName in current) {
       const document = current[operationName];
 
-      let serviceOnly: DocumentNode = removeDirectiveAnnotatedFields(
+      let serviceOnly = removeDirectiveAnnotatedFields(
```

Comment:
@jbaxleyiii I'm happy to walk back the <AST extends ASTNode>` types for `removeDirectiveAnnotatedFields`. With that change, we don't need to define the type here because it's inferred correctly.